src/server/pfs/drive/driver.go

```diff
@@ -18,6 +19,17 @@ import (
 	"google.golang.org/grpc"
 )
 
+func newPermissionError(repo string, commitID string) error {
+	return fmt.Errorf("commit %s/%s has already been finished", repo, commitID)
+}
+
+func IsPermissionError(err error) bool {
+	if readOnly := strings.Contains(err.Error(), "has already been finished"); readOnly {
+		return true
+	}
+	return false
+}
+
```

Comment:
looks like you just want to return `strings.Contains(err.Error(), "has already been finished")`